### ISSUE TITLE
yarn issue with cmd Windows

### ISSUE THREAD

**Issue by Macfly** [graphile/starter#141]
I have see [this PR](https://github.com/graphile/starter/pull/32) that is supposed to fix the setup with Windows.
However when I run it with cmd I have a chmod issue (as the commond does not exist in cmd):
```console
C:\Users\Coco\Documents\projects\plants\starter-2.2.0-cmd\starter-2.2.0>yarn setup
yarn run v1.22.4
$ yarn && node ./scripts/setup.js
[1/4] Resolving packages...
success Already up-to-date.
? What would you like to call your database? graphile_starter
? What's the hostname of your database server (include :port if it's not the default :5432)? postgres:toto@localhost:6543
? Please enter a superuser connection string to the database server (so we can drop/create the 'graphile_starter' and 'graphile_starter_shadow' databases) - IMPORTANT: it must not be a connection to the 'graphile_s
tarter' database itself, instead try 'template1'. postgres://postgres:toto@localhost:6543/template1
'chmod' is not recognized as an internal or external command,
operable program or batch file.
Error: Process exited with status '1' (running 'yarn.cmd server build')
    at spawnSync (C:\Users\Coco\Documents\projects\plants\starter-2.2.0-cmd\starter-2.2.0\scripts\setup.js:46:13)
    at main (C:\Users\Coco\Documents\projects\plants\starter-2.2.0-cmd\starter-2.2.0\scripts\setup.js:265:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I have also tried it with Git bash but still having an issue:
```console
Coco@coco-desk MINGW64 ~/Documents/projects/plants/starter (master)
$ yarn setup
yarn run v1.22.4
$ yarn && node ./scripts/setup.js
[1/4] Resolving packages...
success Already up-to-date.
? What would you like to call your database? graphile_starter
? What's the hostname of your database server (include :port if it's not the default :5432)? postgres:toto@localhost:6543
? Please enter a superuser connection string to the database server (so we can drop/create the 'graphile_starter' and 'graphile_starter_shadow' databases) - IMPORTANT: it must not be a connection to the 'g
raphile_starter' database itself, instead try 'template1'. postgres://postgres:toto@localhost:6543/template1
? We're going to drop (if necessary):

  - database graphile_starter
  - database graphile_starter_shadow
  - database role graphile_starter_visitor (cascade)
  - database role graphile_starter_authenticator (cascade)
  - database role graphile_starter Yes
Installing or reinstalling the roles and database...
C:\Users\Coco\Documents\projects\plants\starter\@app\db\node_modules\.bin\graphile-migrate:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at wrapSafe (internal/modules/cjs/loader.js:1063:16)
    at Module._compile (internal/modules/cjs/loader.js:1111:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
    at Module.load (internal/modules/cjs/loader.js:996:32)
    at Function.Module._load (internal/modules/cjs/loader.js:896:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
Error: Process exited with status '1' (running 'yarn.cmd db reset')
    at spawnSync (C:\Users\Coco\Documents\projects\plants\starter\scripts\setup.js:46:13)
    at main (C:\Users\Coco\Documents\projects\plants\starter\scripts\setup.js:373:3)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Comment by benjie** [comment]
Fixed the first of these (I think) in 93a887c.

The second one seems to be something that npm/yarn/whatever is doing custom on Windows, the code quoted does not seem to exist in the graphile-migrate package.

**Comment by JonParton** [comment]
@Macfly  --> Did you ever solve this and can you share what you did? 

I have the code change from  93a887c in my codebase so it must be something else to fix! 

Thanks and apologies if I have missed some info fixing this elsewhere or failed in my 'googeling' ....

**Comment by Macfly** [comment]
no, couldn't fix the dirname issue.
I use a WSL2 bash to start the project and it works great.

**Comment by JonParton** [comment]
Yeah I'm trying to get it windows compatible as there are a whole host of issues getting some others of my team I wanted to introduce to graphile to use linux (Even WSL) . 

I've swapped out Graphile Migrate for Flyway but I've encountered the same issue with Graphile-Worker so I think I may have to Lean into it and figure out how to fix! I at least have it so I can setup the DB, run the Graphile server and the front end now so just the worker and tests to fix up. 

Had to make a whole host of other changes to the code base too such as escaping quotes in the yarn scripts and alterations to the setup scripts and migrations etc... Once I have something working I'll try and share it as a branch to see if there is any learning to take / any way to make it compatible for both Windows and linux! 

Got to be honest I'm a bit new to the technologies though so I expect it would need a fair amount of review!
